src/components/data/data-table-versions.js

```diff
@@ -0,0 +1,139 @@
+import _ from 'lodash/fp'
+import { Fragment, useEffect, useState } from 'react'
+import { div, h, h1, li, ol, p } from 'react-hyperscript-helpers'
+import { ButtonPrimary, ButtonSecondary, DeleteConfirmationModal, IdContainer, spinnerOverlay } from 'src/components/common'
+import { parseGsUri } from 'src/components/data/data-utils'
+import { icon, spinner } from 'src/components/icons'
+import { TextInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
+import { FormLabel } from 'src/libs/forms'
+import { useCancellation } from 'src/libs/react-utils'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+
+
+const DownloadVersionButton = ({ url }) => {
+  const signal = useCancellation()
+  const [downloadUrl, setDownloadUrl] = useState()
+
+  useEffect(() => {
+    const loadUrl = async () => {
+      try {
+        setDownloadUrl(null)
+        const [bucket, object] = parseGsUri(url)
+        const { url: signedUrl } = await Ajax(signal).DrsUriResolver.getSignedUrl({ bucket, object })
+        setDownloadUrl(signedUrl)
+      } catch (error) {
```

Comment:
I'm going to take your word for it that swallowing the error is the right move.

Reply:
Added error handling in https://github.com/DataBiosphere/terra-ui/pull/3325/commits/39d4ef77e945799c7ab455b94c01c335e82d2bbf.